core/src/main/scala/org/apache/spark/sql/collection/Utils.scala

```diff
@@ -365,6 +365,49 @@ object Utils {
     k
   }
 
+  /**
+   * Utility function to return a metadata for a StructField of StringType, to ensure that the
+   * field is stored (and rendered) as VARCHAR by SnappyStore.
+   * @param md
+   * @param size
+   * @return
+   */
+  def varcharMetadata(md: Metadata = Metadata.empty, size: Int = Constant.MAX_VARCHAR_SIZE):
+  Metadata = {
+    if (size < 1 && size > Constant.MAX_VARCHAR_SIZE) {
+      throw new IllegalArgumentException(s"VARCHAR size should be between 0 " +
+          s"and ${Constant.MAX_VARCHAR_SIZE}")
+    }
+    new MetadataBuilder().withMetadata(md).putString(Constant.CHAR_TYPE_BASE_PROP, "VARCHAR")
+        .putLong(Constant.CHAR_TYPE_SIZE_PROP, size).build()
+  }
+
+  /**
+   * Utility function to return a metadata for a StructField of StringType, to ensure that the
+   * field is stored (and rendered) as CHAR by SnappyStore.
+   * @param md
+   * @param size
+   * @return
+   */
+  def charMetadata(md: Metadata = Metadata.empty, size: Int = Constant.MAX_CHAR_SIZE): Metadata = {
+    if (size < 1 && size > Constant.MAX_CHAR_SIZE) {
+      throw new IllegalArgumentException(s"CHAR size should be between 0 " +
+          s"and ${Constant.MAX_CHAR_SIZE}")
+    }
+    new MetadataBuilder().withMetadata(md).putString(Constant.CHAR_TYPE_BASE_PROP, "CHAR")
+        .putLong(Constant.CHAR_TYPE_SIZE_PROP, size).build()
+  }
+
+  /**
+   * Utility function to return a metadata for a StructField of StringType, to ensure that the
+   * field is rendered as CLOB by SnappyStore.
+   * @param md
+   * @return
+   */
+  def stringMetadata(md: Metadata = Metadata.empty): Metadata = {
+    new MetadataBuilder().withMetadata(md).putString(Constant.CHAR_TYPE_BASE_PROP, "STRING").build()
+  }
+
```

Comment:
Utility methods are needed for CHAR and VARCHAR metadata to allow user to specify these datatypes thru Structype schema. By default, String will return as varchar and can be overridden using the already existing query hint or system var if user wants data in CLOB format. So stringMetadata method may not be needed as user will use query hint when he/she needs data in clob format.